src/Rector/Property/DoctrineTargetEntityStringToClassConstantRector.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Doctrine\Rector\Property;
+
+use Doctrine\ORM\Mapping\Embedded;
+use Doctrine\ORM\Mapping\ManyToMany;
+use Doctrine\ORM\Mapping\ManyToOne;
+use Doctrine\ORM\Mapping\OneToMany;
+use Doctrine\ORM\Mapping\OneToOne;
+use PhpParser\Node;
+use PhpParser\Node\Attribute;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Stmt\Property;
+use Rector\BetterPhpDocParser\PhpDoc\DoctrineAnnotationTagValueNode;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\BetterPhpDocParser\PhpDocParser\ClassAnnotationMatcher;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\ValueObject\PhpVersionFeature;
+use Rector\Doctrine\NodeAnalyzer\AttributeFinder;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Doctrine\Tests\Rector\Property\DoctrineTargetEntityStringToClassConstantRector\DoctrineTargetEntityStringToClassConstantRectorTest
+ */
+final class DoctrineTargetEntityStringToClassConstantRector extends AbstractRector implements MinPhpVersionInterface
+{
+    private const ATTRIBUTE_NAME__TARGET_ENTITY = 'targetEntity';
+
+    private const ATTRIBUTE_NAME__CLASS = 'class';
+
+    /**
+     * @var array<class-string<OneToMany|ManyToOne|OneToOne|ManyToMany|Embedded>, string>
+     */
+    private const VALID_DOCTRINE_CLASSES = [
+        OneToMany::class => self::ATTRIBUTE_NAME__TARGET_ENTITY,
+        ManyToOne::class => self::ATTRIBUTE_NAME__TARGET_ENTITY,
+        OneToOne::class => self::ATTRIBUTE_NAME__TARGET_ENTITY,
+        ManyToMany::class => self::ATTRIBUTE_NAME__TARGET_ENTITY,
+        Embedded::class => self::ATTRIBUTE_NAME__CLASS,
```

Comment:
The keys here must be strings and not `::class` constants. Otherwise the prefixer will turn them into `Rector1234\Doctrine\ORM\...` and they will not be found.